### PR TITLE
[FLINK-12073][table-planner-blink] Support appropriate precision and scale processing of Decimal

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

1.Make Decimal's output more precise than random.
2.Let Decimal process closer to mainstream databases such as Hive/SqlServer.

## Verifying this change

ut

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/typeutils/BigDecimalTypeInfo.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import java.math.BigDecimal;
+import java.util.Arrays;
+
+/**
+ * {@link TypeInformation} for {@link BigDecimal}.
+ *
+ * <p>It differs from {@link BasicTypeInfo#BIG_DEC_TYPE_INFO} in that:
+ * This type includes `precision` and `scale`, similar to SQL DECIMAL.
+ */
+public class BigDecimalTypeInfo extends BasicTypeInfo<BigDecimal> {
```
Comment (KurtYoung) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/typeutils/BigDecimalTypeInfo.java]: Why not calling it `DecimalTypeInfo` ?

Comment (JingsongLi) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/typeutils/BigDecimalTypeInfo.java]: I think it is a `TypeInfo` for `BigDecimal`, so `BigDecimalTypeInfo` is more appropriate.

Comment (godfreyhe) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/typeutils/BigDecimalTypeInfo.java]: Please port BigDecimalTypeInfoTest in this PR

**Comment by godfreyhe** [comment]
LGTM

**Pull request merged**